libcontainerd/container_linux.go

```diff
@@ -191,3 +192,16 @@ func (ctr *container) handleEvent(e *containerd.Event) error {
 	}
 	return nil
 }
+
+func (ctr *container) discardFifos() {
+	for _, i := range []int{syscall.Stdout, syscall.Stderr} {
+		f := ctr.fifo(i)
+		c := make(chan struct{})
+		go func() {
+			close(c) // this channel is used to not close the writer too early, before readonly open has been called.
```

Comment:
If there is no other side then it doesn’t matter what you call first as they both block. If there is something keeping the writer side open then again it doesn’t matter because we don’t need close at all.